k9mail/src/main/java/com/fsck/k9/preferences/DeletePolicy.java

```diff
@@ -0,0 +1,34 @@
+package com.fsck.k9.preferences;
+
+import android.util.Log;
+
+import com.fsck.k9.Account;
+import com.fsck.k9.K9;
+import com.fsck.k9.mail.ServerSettings;
+import com.fsck.k9.mail.store.imap.ImapStore;
+import com.fsck.k9.mail.store.pop3.Pop3Store;
+import com.fsck.k9.mail.store.webdav.WebDavStore;
+
+/**
+ * Defaults for mail deletion.
+ */
+public class DeletePolicy {
```

Comment:
At the moment the deletion policy just depends on the settings store type, so they are coupled there. But I agree,  the to-be-created `AccountCreationService` should probably decide about the policy (or if they are always tied to the store type it would just be a constant).

about the `Account` class, well at some point it has to be done.. unless there's a good reason to keep them as separate constants i would move them to enums.